prober/http.go

```diff
@@ -169,9 +209,14 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 			Help: "Returns the version of HTTP of the probe response",
 		})
 
-		probeFailedDueToRegex = prometheus.NewGauge(prometheus.GaugeOpts{
+		probeFailedDueToHeaderRegex = prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "probe_failed_due_to_header_regex",
```

Comment:
I think one metric for this is fine, the debug logs will always have the full details